pkg/credentials/credentialStorage.go

```diff
@@ -0,0 +1,51 @@
+package credentials
+
+import (
+	"get.porter.sh/porter/pkg/config"
+	"get.porter.sh/porter/pkg/secrets"
+	secretplugins "get.porter.sh/porter/pkg/secrets/pluginstore"
+	crudplugins "get.porter.sh/porter/pkg/storage/pluginstore"
+	"github.com/cnabio/cnab-go/credentials"
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+)
+
+type CredentialsStore = credentials.Store
+type SecretsStore = secrets.Store
+
+var _ CredentialProvider = &CredentialStorage{}
+
+// Credentials provides access to credential sets by instantiating plugins that
+// implement CRUD storage.
+type CredentialStorage struct {
+	*config.Config
+	*CredentialsStore
+	SecretsStore
```

Comment:
This was one of the follow-ups on my todo list/paper. Write some godoc strings in our packages and on the structs to explain the difference between credentials and secrets since it's a new construct not seen in CNAB.

Credential Sets: Mapping from a credential needed by a bundle to where to look for it when the bundle is run. For example: Bundle needs azure storage connection string and it should look for it in an environment variable named X or (new) a key named X.

Secrets: Source of secret values that a tool (porter) can use to resolve credential values (and eventually other secret values that could be needed by other areas of bundles such as parameters or authenticating, etc. For example: When a credential set says that a value can be found in a key named X, porter could look that up in a key/value store and put the value into the "resolved value" in the credential set.

This helps us avoid people copying sensitive data out of secret stores onto the local machines where porter is running and let porter directly access the secrets that the bundle needs and inject them into the credential sets.

Reply:
It helps to look at a credential set file today on disk and realize that the secret data has never been stored in that file, just the mapping of where to find the value.

Reply:
Ah!  Indeed, very helpful explanation.  Making sense to me now.

Reply:
To be clear I'm going to dump this text into a godoc string, I'm just giving it to you first in a GH comment. 😀